src/rtp/RTPPacketizer.cpp

```diff
@@ -0,0 +1,159 @@
+		uint8_t len[2] = {};
+		//Calculate STAP-A aggregation length
+		set2 (len, 0, rtp.GetSize ());
+		//Set it
+		packet->AppendPayload(len, sizeof (len));
+		//Set NAL data
+		packet->AppendPayload(frame->GetData() + rtp.GetPos(), rtp.GetSize());
+	}
+}
+
+std::vector<RTPPacket::shared> RTPPacketizer::GenerateRTPPackets(const std::shared_ptr<MediaFrame>& frame, BYTE codec)
+{
+	std::vector<RTPPacket::shared> rtpPackets;
+	const MediaFrame::RtpPacketizationInfo& info = frame->GetRtpPacketizationInfo();
+
+	for (size_t i=0;i<info.size();i++)
+	{
+		//Get packet
+		const MediaFrame::RtpPacketization& rtp = info[i];
+		//Create rtp packet
+		auto packet = std::make_shared<RTPPacket>(frame->GetType(), codec);
+		//Make sure it is enough length
+		if (rtp.GetTotalLength()>packet->GetMaxMediaLength())
+			//Error
+			continue;
+		//Try to aggreate multiple NALs into a single STAP-A packet for H264
+		if (frame->GetType() == MediaFrame::Video && codec == VideoCodec::H264)
+		{
+			auto last = RTPPacketizer::CalculateAggregationEndIndex(info, i, RTPPAYLOADSIZE);
+			//If we can aggregate more than one packet
+			if (last>i)
+			{
+				RTPPacketizer::AggregatePayloads(frame, packet, i, last);
+				//Skip aggregated packets
+				i = last;
+			} 
+			else 
+			{
+				//Set data
+				packet->SetPayload(frame->GetData() + rtp.GetPos(), rtp.GetSize());
+				//Add prefix
+				packet->PrefixPayload(rtp.GetPrefixData(), rtp.GetPrefixLen());
+			}
+		} 
+		else
+		{
+			//Set data
+			packet->SetPayload(frame->GetData() + rtp.GetPos(), rtp.GetSize());
+			//Add prefix
+			packet->PrefixPayload(rtp.GetPrefixData(), rtp.GetPrefixLen());
+		}
+
+		rtpPackets.push_back(std::move(packet));
+	}
+	return rtpPackets;
+}
+
+void RTPPacketizer::SetPacketsProp(const std::shared_ptr<MediaFrame>& frame, const std::vector<RTPPacket::shared>& rtpPackets, uint64_t lastTimestamp, uint64_t extSeqNum, uint32_t targetBitrateHint)
+{
+	uint32_t rate = 0;
+	RTPPacketizer::VideoFrameInfo videoInfo {0};
+	bool addVideoLayerInfo = false;
+
+	switch(frame->GetType())
+	{
+		case MediaFrame::Audio:
+		{
+			//get audio frame
+			auto audioFrame = std::static_pointer_cast<AudioFrame>(frame);
+			//Set correct clock rate for audio codec
+			rate = AudioCodec::GetClockRate(audioFrame->GetCodec());
+			break;
+		}
+		case MediaFrame::Video:
+		{
+			//get Video frame
+			auto videoFrame = std::static_pointer_cast<VideoFrame>(frame);
+			rate = 90000;
+			videoInfo.targetBitrate = videoFrame->GetTargetBitrate() ? videoFrame->GetTargetBitrate() : targetBitrateHint;
+			videoInfo.targetFps = videoFrame->GetTargetFps();
+			addVideoLayerInfo = videoFrame->IsIntra() && videoInfo.targetBitrate;
+			if (videoFrame->GetWidth() != 0 && videoFrame->GetHeight() != 0)
+			{
+				videoInfo.width = videoFrame->GetWidth();
+				videoInfo.height = videoFrame->GetHeight();
+			}
+			break;
+		}
+		default:
+			break;
+		
+	}
+
+	for (size_t i=0; i<rtpPackets.size(); i++)
+	{
+		auto packet = rtpPackets[i];
```

Comment:
Do not copy, it will cause a new shared pointer to be created, degrading performance
```suggestion
		auto& packet = rtpPackets[i];
```